src/pint/plot_utils.py

```diff
@@ -176,3 +177,125 @@ def phaseogram_binned(
+        If False, the priors will be plotted independent of the histograms.
+        In certain cases, such as broad priors, the priors or histograms
+        might not be apparent on the same plot due to one being significantly
+        larger than the other. The scaling is for visual purposes to clearly 
+        plot the priors with the samples
+    plotfile : optional
+        Name of the plotfile to be saved
+    """
+    keys = []
+    values = []
+    for k, v in chains.items():
+        keys.append(k), values.append(v)
+
+    priors = []
+    x_range = []
+    counts = []
+    for i in range(0, len(keys[:-1])):
+        values[i] = values[i][burnin:].flatten()
+        x_range.append(np.linspace(values[i].min(), values[i].max(), num=bins))
+        priors.append(getattr(model, keys[i]).prior.pdf(x_range[i]))
+        a, x = np.histogram(values[i], bins=bins, density=True)
+        counts.append(a)
+
+    fig, axs = plt.subplots(len(keys), figsize=(8, 11), constrained_layout=True)
+
+    for i, p in enumerate(keys):
+        if i != len(keys[:-1]):
+            axs[i].set_xlabel(
+                str(p)
+                + ": Mean Value = "
+                + str("{:.9e}".format(values[i].mean()))
+                + " ("
+                + str(getattr(model, p).units)
+                + ")"
+            )
+            axs[i].axvline(
+                -2 * values[i].std(), color="b", linestyle="--", label="2 sigma"
+            )
+            axs[i].axvline(2 * values[i].std(), color="b", linestyle="--")
+            axs[i].hist(
+                values[i] - values[i].mean(), bins=bins, density=True, label="Samples"
+            )
+            if scale:
+                axs[i].plot(
+                    x_range[i] - values[i].mean(),
+                    priors[i] * counts[i].max() / priors[i].max(),
+                    label="Prior Probability",
+                    color="g",
+                )
+            else:
+                axs[i].plot(
+                    x_range[i] - values[i].mean(),
+                    priors[i],
+                    label="Prior Probability",
+                    color="g",
+                )
+            if maxpost_fitvals is not None:
+                axs[i].axvline(
+                    maxpost_fitvals[i] - values[i].mean(),
+                    color="c",
+                    linestyle="--",
+                    label="Maximum Likelihood Value",
+                )
+            if fitvals is not None:
+                axs[i].axvline(
+                    fitvals[i] - values[i].mean(),
+                    color="m",
+                    linestyle="--",
+                    label="Original Parameter Fit Value",
+                )
+        else:
+            handles, labels = axs[0].get_legend_handles_labels()
+            axs[i].set_axis_off()
+            axs[i].legend(handles, labels)
+    if plotfile is not None:
+        plt.savefig(plotfile)
+        plt.close()
+    else:
+        plt.show()
```

Comment:
Definitely not a good idea to include a `plt.show()`; what if users want to annotate the plots?